Sources/W3CTraceContext/TraceID.swift

```diff
@@ -52,101 +58,151 @@ public struct TraceID: Sendable {
+                lhs._bytes.6 == rhs._bytes.6 &&
+                lhs._bytes.7 == rhs._bytes.7 &&
+                lhs._bytes.8 == rhs._bytes.8 &&
+                lhs._bytes.9 == rhs._bytes.9 &&
+                lhs._bytes.10 == rhs._bytes.10 &&
+                lhs._bytes.11 == rhs._bytes.11 &&
+                lhs._bytes.12 == rhs._bytes.12 &&
+                lhs._bytes.13 == rhs._bytes.13 &&
+                lhs._bytes.14 == rhs._bytes.14 &&
+                lhs._bytes.15 == rhs._bytes.15
+        }
+
+        public func hash(into hasher: inout Hasher) {
+            hasher.combine(_bytes.0)
+            hasher.combine(_bytes.1)
+            hasher.combine(_bytes.2)
+            hasher.combine(_bytes.3)
+            hasher.combine(_bytes.4)
+            hasher.combine(_bytes.5)
+            hasher.combine(_bytes.6)
+            hasher.combine(_bytes.7)
+            hasher.combine(_bytes.8)
+            hasher.combine(_bytes.9)
+            hasher.combine(_bytes.10)
+            hasher.combine(_bytes.11)
+            hasher.combine(_bytes.12)
+            hasher.combine(_bytes.13)
+            hasher.combine(_bytes.14)
+            hasher.combine(_bytes.15)
+        }
     }
 }
 
-extension TraceID: Hashable {
-    public func hash(into hasher: inout Hasher) {
-        hasher.combine(_bytes.0)
-        hasher.combine(_bytes.1)
-        hasher.combine(_bytes.2)
-        hasher.combine(_bytes.3)
-        hasher.combine(_bytes.4)
-        hasher.combine(_bytes.5)
-        hasher.combine(_bytes.6)
-        hasher.combine(_bytes.7)
-        hasher.combine(_bytes.8)
-        hasher.combine(_bytes.9)
-        hasher.combine(_bytes.10)
-        hasher.combine(_bytes.11)
-        hasher.combine(_bytes.12)
-        hasher.combine(_bytes.13)
-        hasher.combine(_bytes.14)
-        hasher.combine(_bytes.15)
+extension TraceID.Bytes: CustomStringConvertible {
+    /// A 32-character hex string representation of the bytes.
+    public var description: String {
+        String(decoding: hexBytes, as: UTF8.self)
+    }
+
+    /// A 32-character UTF-8 hex byte array representation of the bytes.
+    public var hexBytes: [UInt8] {
+        var asciiBytes = [UInt8](repeating: 0, count: 32)
+        for i in startIndex ..< endIndex {
+            let byte = self[i]
+            asciiBytes[2 * i] = Hex.lookup[Int(byte >> 4)]
+            asciiBytes[2 * i + 1] = Hex.lookup[Int(byte & 0x0F)]
+        }
+        return asciiBytes
     }
 }
 
+extension TraceID: Equatable {}
+
+extension TraceID: Hashable {}
+
 extension TraceID: Identifiable {
-    public var id: [UInt8] { bytes }
+    public var id: Self { self }
 }
 
 extension TraceID: CustomStringConvertible {
-    /// A 32 character hex string representation of the span ID.
+    /// A 32-character hex string representation of the trace ID.
     public var description: String {
-        String(decoding: hexBytes, as: UTF8.self)
+        String(decoding: bytes.hexBytes, as: UTF8.self)
```

Comment:
```suggestion
        String(describing: bytes)
```
Same suggestion as with `SpanID`.